src/luarocks/util.lua

```diff
@@ -587,14 +587,25 @@ do
       if luaver and luaver ~= lv then
          return nil
       end
+      return lv
+   end
+
+   function util.get_luajit_version()
+      local cfg = require("luarocks.core.cfg")
+      if cfg.cache.luajit_version_checked then
+         return cfg.cache.luajit_version 
+      end
+      cfg.cache.luajit_version_checked = true
+
       local ljv
-      if lv == "5.1" then
-         ljv = util.popen_read(Q(lua_exe) .. ' -e "io.write(tostring(jit and jit.version:sub(8)))"')
+      if cfg.lua_version == "5.1" then
+         ljv = util.popen_read(Q(cfg.variables["LUA_BINDIR"] .. "/" .. cfg.lua_interpreter) .. ' -e "io.write(tostring(jit and jit.version:sub(8)))"')
```

Comment:
Shouldn't/couldn't this be done always (regardless of `lua_version`)?

Reply:
LuaRocks generally assumes that LuaJIT implies Lua 5.1, to avoid trouble with different LuaJITs compiled with different levels of Lua 5.x compatibility.